app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt

```diff
@@ -0,0 +1,388 @@
+                performBulkOperation(
+                    type = bulkOperationType,
+                    selectedFilesCount = getSelectedFilesCount(),
+                    destinationFolder = File(id = folderId, name = folderName, driveId = AccountUtils.currentDriveId),
+                )
+            }
+        }
+    }
+
+    fun openMultiSelect() {
+        multiSelectManager.isMultiSelectOpened = true
+
+        adapter?.apply { notifyItemRangeChanged(0, itemCount) }
+
+        multiSelectToolbar?.isGone = true
+        multiSelectLayout?.root?.isVisible = true
+    }
+
+    fun onItemSelected(selectedNumber: Int? = null) {
+        val fileSelectedNumber = selectedNumber ?: multiSelectManager.getValidSelectedItems().size
+
+        when (fileSelectedNumber) {
+            0, 1 -> {
+                val isEnabled = fileSelectedNumber == 1
+                enableMultiSelectButtons(isEnabled)
+            }
+        }
+
+        multiSelectLayout?.titleMultiSelect?.text = resources.getQuantityString(
+            R.plurals.fileListMultiSelectedTitle, fileSelectedNumber, fileSelectedNumber
+        )
+    }
+
+    fun enableMultiSelectButtons(isEnabled: Boolean) {
+        multiSelectLayout?.apply {
+            deleteButtonMultiSelect.isEnabled = isEnabled
+            moveButtonMultiSelect.isEnabled = isEnabled
+            menuButtonMultiSelect.isEnabled = isEnabled
+        }
+    }
+
+    open fun closeMultiSelect() {
+        multiSelectManager.apply {
+            selectedItems = RealmList()
+            areAllSelected = false
+            isMultiSelectOpened = false
+        }
+
+        adapter?.apply { notifyItemRangeChanged(0, itemCount) }
+
+        multiSelectToolbar?.isVisible = true
+        multiSelectLayout?.root?.isGone = true
+    }
+
+    abstract fun getSelectedFilesCount(): Int?
+
+    fun moveFiles(folderId: Int?) {
+        requireContext().moveFileClicked(folderId, selectFolderResultLauncher)
+    }
+
+    fun deleteFiles(fileCount: Int? = null) {
+        performBulkOperation(type = BulkOperationType.TRASH, selectedFilesCount = fileCount)
+    }
+
+    fun colorFolders() {
+        if (AccountUtils.getCurrentDrive()?.pack == Drive.DrivePack.FREE.value) {
+            safeNavigate(R.id.colorFolderUpgradeBottomSheetDialog)
+        } else {
+            safeNavigate(FileListFragmentDirections.actionFileListToColorFolder(null))
+        }
+    }
+
+    fun duplicateFiles() {
+        val intent = Intent(requireContext(), SelectFolderActivity::class.java).apply {
+            putExtra(USER_ID_TAG, AccountUtils.currentUserId)
+            putExtra(USER_DRIVE_ID_TAG, AccountUtils.currentDriveId)
+            putExtra(CUSTOM_ARGS_TAG, bundleOf(BULK_OPERATION_CUSTOM_TAG to BulkOperationType.COPY))
+        }
+        selectFolderResultLauncher.launch(intent)
+    }
+
+    open fun performBulkOperation(
+        type: BulkOperationType,
+        areAllFromTheSameFolder: Boolean = true,
+        selectedFilesCount: Int? = null,
```

Comment:
You have to explain to me what it is used for, knowing that after we call `getValidSelectedItems `
The only use I see is for the "all selected", so at that point rename it everywhere.